Sourcery/Models/TypeName.swift

```diff
@@ -192,6 +209,45 @@ public final class TypeName: NSObject, AutoCoding, AutoEquatable, AutoDiffable,
 
 }
 
+/// Descibes Swift class
+public final class GenericType: NSObject, SourceryModel {
+    /// the name of the base type.
+    ///
+    /// `Array<Int>`'s GenericType.name is `Array`
+    public let name: String
```

Comment:
I think we should also provide `Type` property that will reference to this generic type definition.